### PR TITLE
Added Cart functionality

### PR DESCRIPTION
**Pull request by psychoactiv**
-Added cart functionality
-Added Snackbar for Warnings
27 03 2022_02 43 17_REC

### CODE REVIEW

File: src/components/Best-selling-product/best-selling.jsx [hunk 1]
```diff
@@ -35,7 +35,9 @@ export function BestSelling() {
                   <span className={`${bg} same-star chng-pad`}>
                     {item.rating} <i className={`fas fa-star ${font}`}></i>
                   </span>
-                  <span className="font-wt-black card-price">{item.price}</span>
+                  <span className="font-wt-black card-price">
+                    {item.price.toLocaleString("en-US")}
```
Comment (JantuDeb) [inline src/components/Best-selling-product/best-selling.jsx:39]: `en-IN`

File: src/components/cart-checkout-card/cart-checkout-card.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    const getDetails = cartState.reduce(
+      (acum, cur) => {
+        acum.CartQty += cur.qty;
+        acum.totalMrp += cur.price * cur.qty;
+        const [discountedPrice, sumToMinus] = calculateCartItems(cur.price, 40);
+        acum.discountedPrice += sumToMinus * cur.qty;
+        acum.subTotal += discountedPrice * cur.qty;
+        return acum;
+      },
+      { CartQty: 0, totalMrp: 0, discountedPrice: 0, subTotal: 0 }
```
Comment (JantuDeb) [inline src/components/cart-checkout-card/cart-checkout-card.jsx:19]: You can directly call this function no need to use useEffect and state

**Review by JantuDeb — APPROVED** [review]

File: src/components/cart-checkout-card/cart-checkout-card.jsx (hunk 2 above)
Comment (psychoactiv) [reply src/components/cart-checkout-card/cart-checkout-card.jsx:19]: Sure, will do that change

File: src/components/Filter/Filter.jsx [hunk 3]
```diff
@@ -20,20 +20,25 @@ export function Filter() {
       dispatch({ type: "GPU" });
     checkIsTrue("Pixel perfect Camera for memories", "categoiries") &&
```
Comment (tusharkandpal) [inline src/components/Filter/Filter.jsx:21]: typo in `categories`

File: src/components/Filter/Filter.jsx [hunk 4]
```diff
@@ -20,20 +20,25 @@ export function Filter() {
...
       dispatch({ type: "CAMERA" });
-    // checkIsTrue() &&
+    checkIsTrue("Apple", "brand") &&
+      dispatch({ type: "BRAND", payload: "APPLE" });
+    checkIsTrue("OnePlus", "brand") &&
+      dispatch({ type: "BRAND", payload: "ONEPLUS" });
+    checkIsTrue("Samsung", "brand") &&
+      dispatch({ type: "BRAND", payload: "SAMSUNG" });
```
Comment (tusharkandpal) [inline src/components/Filter/Filter.jsx:28]: i would suggest you to extract it out in a different function.

File: src/utils/cart-reducer-fn.jsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        };
+        return [...newStateArr];
+      } else if (state[foundIndex].qty === 1) {
+        newStateArr.splice(foundIndex, 1);
+      }
+      return [...newStateArr];
+    case "REMOVE_ITEM":
+      const findItemIndex = state.findIndex((item) => item.id === payload);
+      const stateArr = [...state];
+      stateArr.splice(findItemIndex, 1);
```
Comment (tusharkandpal) [inline src/utils/cart-reducer-fn.jsx:42]: you can combine these two steps in one `[...state].splice`

File: src/utils/filterfn.jsx [hunk 6]
```diff
@@ -108,7 +109,13 @@ function ratings(state, data) {
...
-  if (state.SAMSUNG || state.APPLE || state.VIVO || state.ASUS) {
+  if (
+    state.SAMSUNG ||
+    state.APPLE ||
+    state.VIVO ||
+    state.ASUS ||
+    state.ONEPLUS
+  ) {
     const newArrData = [...data];
     return newArrData.filter((item) => state[item.brand.toUpperCase()]);
```
Comment (tusharkandpal) [inline src/utils/filterfn.jsx:120]: here i think you don't need to spread as filter will return a new array itself.

**Review by tusharkandpal — APPROVED** [review]
your function names are quite creative. plus functionality is good so far.!

File: src/components/Filter/Filter.jsx (hunk 3 above)
Comment (psychoactiv) [reply src/components/Filter/Filter.jsx:21]: Thanks didn't see that one, just saw that my camera category was not working 😅

**Pull request merged**